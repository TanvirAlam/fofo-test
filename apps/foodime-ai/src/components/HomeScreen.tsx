import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet, Platform } from 'react-native';
import { useNavigation } from '@react-navigation/native';

const HomeScreen = () => {
  const navigation = useNavigation();

  const navigateToChat = () => {
    // @ts-expect-error - Navigation types not properly configured
    navigation.navigate('Chat');
  };

  const navigateToSettings = () => {
    // @ts-expect-error - Navigation types not properly configured
    navigation.navigate('Settings');
  };

  const navigateToAITesting = () => {
    // @ts-expect-error - Navigation types not properly configured
    navigation.navigate('AITesting');
  };

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>RFLCT AI</Text>
        <Text style={styles.subtitle}>Speech-to-Speech AI Assistant</Text>
      </View>

      <View style={styles.buttonContainer}>
        <TouchableOpacity 
          style={[styles.button, styles.primaryButton]} 
          onPress={navigateToChat}
        >
          <Text style={styles.primaryButtonText}>Start AI Chat</Text>
        </TouchableOpacity>

        <TouchableOpacity 
          style={[styles.button, styles.secondaryButton]} 
          onPress={navigateToSettings}
        >
          <Text style={styles.secondaryButtonText}>Settings</Text>
        </TouchableOpacity>

        <TouchableOpacity 
          style={[styles.button, styles.testingButton]} 
          onPress={navigateToAITesting}
        >
          <Text style={styles.testingButtonText}>ðŸ§ª AI Testing Suite</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.features}>
        <Text style={styles.featuresTitle}>Features:</Text>
        <Text style={styles.featureItem}>â€¢ Multiple AI providers (OpenAI, Claude, Gemini)</Text>
        <Text style={styles.featureItem}>â€¢ Advanced TTS with 8 providers (ElevenLabs, Hugging Face, OpenAI, etc.)</Text>
        <Text style={styles.featureItem}>â€¢ Multi-provider STT (Whisper, Google, Azure, Deepgram)</Text>
        <Text style={styles.featureItem}>â€¢ Complete order processing system</Text>
        <Text style={styles.featureItem}>â€¢ Cross-platform support (Web & Mobile)</Text>
        <Text style={styles.featureItem}>â€¢ 4-digit RFLCT code support</Text>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f8f9fa',
    padding: 20,
    justifyContent: 'center',
  },
  header: {
    alignItems: 'center',
    marginBottom: 50,
  },
  title: {
    fontSize: 32,
    fontWeight: 'bold',
    color: '#2c3e50',
    marginBottom: 10,
  },
  subtitle: {
    fontSize: 18,
    color: '#7f8c8d',
    textAlign: 'center',
  },
  buttonContainer: {
    marginBottom: 40,
  },
  button: {
    padding: 15,
    borderRadius: 10,
    marginBottom: 15,
    alignItems: 'center',
    ...(Platform.OS === 'web' && {
      cursor: 'pointer',
    }),
  },
  primaryButton: {
    backgroundColor: '#3498db',
  },
  secondaryButton: {
    backgroundColor: 'transparent',
    borderWidth: 2,
    borderColor: '#3498db',
  },
  testingButton: {
    backgroundColor: '#e74c3c',
  },
  primaryButtonText: {
    color: '#ffffff',
    fontSize: 18,
    fontWeight: 'bold',
  },
  secondaryButtonText: {
    color: '#3498db',
    fontSize: 18,
    fontWeight: 'bold',
  },
  testingButtonText: {
    color: '#ffffff',
    fontSize: 18,
    fontWeight: 'bold',
  },
  features: {
    backgroundColor: '#ffffff',
    padding: 20,
    borderRadius: 10,
    ...(Platform.OS === 'web' && {
      boxShadow: '0 2px 4px rgba(0,0,0,0.1)',
    }),
    elevation: 2,
  },
  featuresTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#2c3e50',
    marginBottom: 15,
  },
  featureItem: {
    fontSize: 16,
    color: '#34495e',
    marginBottom: 8,
    lineHeight: 24,
  },
});

export default HomeScreen;
