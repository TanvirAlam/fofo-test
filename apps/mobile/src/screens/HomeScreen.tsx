import React from "react";
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableOpacity,
  Alert,
  Button,
} from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { useTranslation } from "react-i18next";
import { COLORS, typography } from "../theme";
import { useNavigation } from "@react-navigation/native";

export function HomeScreen() {
  const handlePress = () => {
    Alert.alert("Success!", "Your Foodime mobile app is working perfectly! ðŸŽ‰");
  };

  const { t, i18n } = useTranslation();

  const toggleLanguage = () => {
    const nextLang = i18n.language === "en" ? "da" : "en";
    i18n.changeLanguage(nextLang);
  };

  const navigation = useNavigation();

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView contentInsetAdjustmentBehavior="automatic">
        <View style={styles.content}>
          <Text style={[typography.Heading1, styles.title]}>
            Welcome to Foodime!
          </Text>
          <Text style={[typography.Heading4, styles.subtitle]}>
            Your mobile app is ready
          </Text>
          <Text style={[typography.Body_Large, styles.description]}>
            This is your React Native app with Expo, integrated into the Foodime
            monorepo.
          </Text>

          <View style={styles.features}>
            <Text style={[typography.Heading4, styles.featuresTitle]}>
              Features included:
            </Text>
            <Text style={[typography.Body_Large, styles.feature]}>
              â€¢ React Navigation
            </Text>
            <Text style={[typography.Body_Large, styles.feature]}>
              â€¢ Safe Area handling
            </Text>
            <Text style={[typography.Body_Large, styles.feature]}>
              â€¢ TypeScript support
            </Text>
            <Text style={[typography.Body_Large, styles.feature]}>
              â€¢ ESLint configuration
            </Text>
            <Text style={[typography.Body_Large, styles.feature]}>
              â€¢ Shared UI components
            </Text>
            <Text style={[typography.Body_Large, styles.feature]}>
              â€¢ Monorepo integration
            </Text>
          </View>

          <TouchableOpacity style={styles.button} onPress={handlePress}>
            <Text style={[typography.Heading4, styles.buttonText]}>
              Test App ðŸš€
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.button}
            onPress={() => navigation.navigate("Register" as never)}
          >
            <Text style={[typography.Heading4, styles.buttonText]}>
              {t("goToRegister")}
            </Text>
          </TouchableOpacity>

          <View style={styles.text}>
            <Button
              title={t("changeLanguage") || "Change Language"}
              onPress={toggleLanguage}
            />
          </View>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: COLORS.PRIMARY.WHITE,
  },
  content: {
    padding: 20,
    alignItems: "center",
  },
  title: {
    color: COLORS.SECONDARY.DARK_GRAY,
    marginBottom: 10,
    textAlign: "center",
  },
  subtitle: {
    color: COLORS.PRIMARY.GRAY,
    marginBottom: 20,
    textAlign: "center",
  },
  description: {
    color: COLORS.SECONDARY.MEDIUM_GRAY,
    textAlign: "center",
    marginBottom: 30,
    lineHeight: 24,
  },
  features: {
    alignSelf: "stretch",
    backgroundColor: COLORS.SECONDARY.LIGHT_GRAY,
    padding: 20,
    borderRadius: 10,
  },
  featuresTitle: {
    color: COLORS.SECONDARY.DARK_GRAY,
    marginBottom: 15,
  },
  feature: {
    color: COLORS.SECONDARY.MEDIUM_GRAY,
    marginBottom: 8,
    lineHeight: 22,
  },
  button: {
    backgroundColor: COLORS.PRIMARY.BLUE,
    paddingHorizontal: 30,
    paddingVertical: 15,
    borderRadius: 25,
    marginTop: 30,
    shadowColor: COLORS.PRIMARY.BLACK,
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  buttonText: {
    color: COLORS.PRIMARY.WHITE,
    fontSize: 18,
    fontWeight: "600",
    textAlign: "center",
  },
  text: {
    margin: 30,
    textAlign: "center",
  },
});
