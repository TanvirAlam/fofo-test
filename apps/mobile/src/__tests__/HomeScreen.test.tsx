import React from 'react';
import { Alert } from 'react-native';
import { render, fireEvent, screen } from '../test/test-utils';
import { HomeScreen } from '../screens/HomeScreen';

// Mock Alert.alert
const mockAlert = Alert.alert as jest.Mock;

describe('HomeScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with all expected elements', () => {
    render(<HomeScreen />);

    // Check if main title is present
    expect(screen.getByText('Welcome to Foodime!')).toBeTruthy();
    
    // Check if subtitle is present
    expect(screen.getByText('Your mobile app is ready')).toBeTruthy();
    
    // Check if description is present
    expect(screen.getByText(/This is your React Native app with Expo/)).toBeTruthy();
    
    // Check if features section is present
    expect(screen.getByText('Features included:')).toBeTruthy();
    
    // Check if test button is present
    expect(screen.getByText('Test App ðŸš€')).toBeTruthy();
  });

  it('displays all feature items', () => {
    render(<HomeScreen />);

    const expectedFeatures = [
      'â€¢ React Navigation',
      'â€¢ Safe Area handling',
      'â€¢ TypeScript support',
      'â€¢ ESLint configuration',
      'â€¢ Shared UI components',
      'â€¢ Monorepo integration',
    ];

    expectedFeatures.forEach(feature => {
      expect(screen.getByText(feature)).toBeTruthy();
    });
  });

  it('shows success alert when test button is pressed', () => {
    render(<HomeScreen />);

    const testButton = screen.getByText('Test App ðŸš€');
    
    fireEvent.press(testButton);

    expect(mockAlert).toHaveBeenCalledWith(
      'Success!',
      'Your Foodime mobile app is working perfectly! ðŸŽ‰'
    );
    expect(mockAlert).toHaveBeenCalledTimes(1);
  });

  it('has accessible button', () => {
    render(<HomeScreen />);

    const testButton = screen.getByText('Test App ðŸš€');
    expect(testButton).toBeTruthy();
    
    // Test that the button is touchable
    fireEvent.press(testButton);
    expect(mockAlert).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<HomeScreen />);
    expect(toJSON()).toMatchSnapshot();
  });
});
