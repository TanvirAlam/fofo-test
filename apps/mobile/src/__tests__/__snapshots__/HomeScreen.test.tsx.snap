// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomeScreen matches snapshot 1`] = `
<NavigationContainer>
  <ScrollView
    contentInsetAdjustmentBehavior="automatic"
  >
    <View
      style={
        {
          "alignItems": "center",
          "padding": 20,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Poppins_Bold",
              "fontSize": 28,
            },
            {
              "color": "rgba(51, 51, 51, 1)",
              "marginBottom": 10,
              "textAlign": "center",
            },
          ]
        }
      >
        Welcome to Foodime!
      </Text>
      <Text
        style={
          [
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Poppins_Bold",
              "fontSize": 18,
            },
            {
              "color": "rgba(158, 158, 158, 1)",
              "marginBottom": 20,
              "textAlign": "center",
            },
          ]
        }
      >
        Your mobile app is ready
      </Text>
      <Text
        style={
          [
            {
              "color": "rgba(255, 255, 255, 1)",
              "fontFamily": "Poppins_Regular",
              "fontSize": 16,
            },
            {
              "color": "rgba(68, 68, 68, 1)",
              "lineHeight": 24,
              "marginBottom": 30,
              "textAlign": "center",
            },
          ]
        }
      >
        This is your React Native app with Expo, integrated into the Foodime monorepo.
      </Text>
      <View
        style={
          {
            "alignSelf": "stretch",
            "backgroundColor": "rgba(245, 245, 245, 1)",
            "borderRadius": 10,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Bold",
                "fontSize": 18,
              },
              {
                "color": "rgba(51, 51, 51, 1)",
                "marginBottom": 15,
              },
            ]
          }
        >
          Features included:
        </Text>
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Regular",
                "fontSize": 16,
              },
              {
                "color": "rgba(68, 68, 68, 1)",
                "lineHeight": 22,
                "marginBottom": 8,
              },
            ]
          }
        >
          â€¢ React Navigation
        </Text>
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Regular",
                "fontSize": 16,
              },
              {
                "color": "rgba(68, 68, 68, 1)",
                "lineHeight": 22,
                "marginBottom": 8,
              },
            ]
          }
        >
          â€¢ Safe Area handling
        </Text>
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Regular",
                "fontSize": 16,
              },
              {
                "color": "rgba(68, 68, 68, 1)",
                "lineHeight": 22,
                "marginBottom": 8,
              },
            ]
          }
        >
          â€¢ TypeScript support
        </Text>
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Regular",
                "fontSize": 16,
              },
              {
                "color": "rgba(68, 68, 68, 1)",
                "lineHeight": 22,
                "marginBottom": 8,
              },
            ]
          }
        >
          â€¢ ESLint configuration
        </Text>
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Regular",
                "fontSize": 16,
              },
              {
                "color": "rgba(68, 68, 68, 1)",
                "lineHeight": 22,
                "marginBottom": 8,
              },
            ]
          }
        >
          â€¢ Shared UI components
        </Text>
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Regular",
                "fontSize": 16,
              },
              {
                "color": "rgba(68, 68, 68, 1)",
                "lineHeight": 22,
                "marginBottom": 8,
              },
            ]
          }
        >
          â€¢ Monorepo integration
        </Text>
      </View>
      <TouchableOpacity
        onPress={[Function]}
        style={
          {
            "backgroundColor": "rgba(0, 122, 255, 1)",
            "borderRadius": 25,
            "elevation": 5,
            "marginTop": 30,
            "paddingHorizontal": 30,
            "paddingVertical": 15,
            "shadowColor": "rgba(0, 0, 0, 1)",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Bold",
                "fontSize": 18,
              },
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          Test App ðŸš€
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        onPress={[Function]}
        style={
          {
            "backgroundColor": "rgba(0, 122, 255, 1)",
            "borderRadius": 25,
            "elevation": 5,
            "marginTop": 30,
            "paddingHorizontal": 30,
            "paddingVertical": 15,
            "shadowColor": "rgba(0, 0, 0, 1)",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontFamily": "Poppins_Bold",
                "fontSize": 18,
              },
              {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          goToRegister
        </Text>
      </TouchableOpacity>
      <View
        style={
          {
            "margin": 30,
            "textAlign": "center",
          }
        }
      >
        <Button
          onPress={[Function]}
          title="changeLanguage"
        />
      </View>
    </View>
  </ScrollView>
</NavigationContainer>
`;
